### DISCUSSION TITLE
请教一下，关于Xray(M+B+D+G+A)+Caddy(N+T)的配置

### DISCUSSION THREAD

**Discussion by wclmgcd** [lxhao61/integrated-examples#180] [Q&A] [answered]
在自己的vps上，搭建有伪装网站www.11111.tk,  同时使用docker-compose搭建有reader.11111.tk（阅读服务器版），rss.11111.tk(miniflux),    rsshub.11111.tk(rsshub)这三个网站，还有v2ray和NaiveProxy这两个服务，用的是caddy1，现在因为需要把v2ray改成Xray的VLESS+Vision+REALITY ，只有升级到caddy 2，看了大佬的配置，有一些疑问咨询：

1.我看大佬caddy的配置中，VLESS+Vision+REALITY 仅使用 z1.xx.yy 域名，NaiveProxy 仅使用 z3.xx.yy 域名，不可以都使用www.11111.tk这一个域名吗？毕竟VLESS+Vision+REALITY也是偷自己的证书。
2.  z1.xx.yy， z3.xx.yy是同一个域名的xx.yy的二级域名？还是完全不同的域名？  
3. 我看证书申请这里
```
    "tls": {
      "certificates": {
        "automate": ["z1.xx.yy","z2.xx.yy","z3.xx.yy"] //自动化管理 TLS 证书（包括获取与更新证书）。修改为自己的域名。
      },
      "automation": {
        "policies": [{
          "issuers": [{
            "module": "acme",
            "email": "your@email.com" //修改为自己的电子邮箱，与下一致。（选配）
          },
          {
            "module": "zerossl",
            "email": "your@email.com" //修改为自己的电子邮箱，与上一致。（选配）
          }]
        }]
      }
    }
```
acme和zerossl是不是只用一个，或者如何指定z1.xx.yy这个用acme申请，z3.xx.yy用zerossl申请？

**Comment by lxhao61** [comment, marked as answer]
1、示例中 z1.xx.yy、z3.xx.yy  表示使用两个不同域名来对应应用。当然你 VLESS+Vision+REALITY 与 NaiveProxy 可以使用同一域名，但这样  NaiveProxy 就有  TLS in TLS 问题了（被封风险）。
2、本项目示例中除非使用通配符证书，否则所有不同域名可以是同一个域名的 xx.yy 的二级域名，也可以是完全不同的域名。（随意，不影响。）
3、若不需要固定证书路径（外部引用需要），推荐 acme 和 zerossl 都配置（单一证书机构出问题不会影响证书申请）。指定 z1.xx.yy 用 acme 申请，z3.xx.yy 用 zerossl 申请见如下（不推荐）:
```
      "automation": {
        "policies": [{
          "subjects": ["z1.xx.yy","z2.xx.yy"], //限定域名，修改为自己的。
          "issuers": [{
            "module": "acme",
            "email": "your@email.com" //修改为自己的电子邮箱。（选配）
          }]
        },
        {
          "subjects": ["z3.xx.yy"], //限定域名，修改为自己的。
          "issuers": [{
            "module": "zerossl",
            "email": "your@email.com" //修改为自己的电子邮箱。（选配）
          }]
        }]
      }
```

**Comment by wclmgcd** [comment]
感谢大佬解答，还有两个问题：
1.你说“VLESS+Vision+REALITY 与 NaiveProxy 可以使用同一域名，但这样 NaiveProxy 就有 TLS in TLS 问题了”，这里我原来是按照美博园的教程搭建的，配置文件如下：
```
......
forwardproxy {
    basicauth Mute3774 vNPR9DQuXTyqZC!
    hide_ip
    hide_via
    probe_resistance
    upstream http://127.0.0.1:8080
    }
}
......
```
我看大佬的caddy自带了NaiveProxy，不在另外安装NaiveProxy，那如果我用美博园这种在caddy2中，再另外安装NaiveProxy可不可以？是不是就能解决与REALITY同一个域名的TLS in TLS 问题？ 

2. 不同二级域名的转发，再caddy2里如何写？
在caddy 1中，我的格式如下：
```
reader.11111.tk {
    proxy / 127.0.0.1:4396
    gzip
}
```

**Comment by lxhao61** [comment]
1、你按照美博园教程的配置一样有 TLS in TLS，跟是否 Caddy1 的 NaiveProxy 或 Caddy2 的 NaiveProxy 无关，仅跟配置有关（当然你 
 Caddy1 配置不支持本示例 Caddy2 配置）！另外你 Caddy1 配置已过时早就不推荐了，参考本示例配置即可。
2、不同二级域名的转发见下（gzip 与 proxy 配置）：
```
......
            "handle": [{
              "handler": "encode",
              "encodings": {
                "gzip": {
                  "level": 4
                }
              }
            },
            {
              "handler": "reverse_proxy",
              "upstreams": [{
                "dial": "127.0.0.1:4396"
              }]
            }]
......
```
